docs/advanced.rst

```diff
@@ -562,3 +562,79 @@ or the record invocation: ``app="SomeAppName"``.
         opinion = models.CharField(max_length=2000)
 
     register(Opinion, app="SomeAppName")
+
+Custom History Model Name
+-------------------------
+
+By default the history model names are the base model name prefixed with "Historical".
+So given a model named ``Poll``, the resultant history class will be named ``HistoricalPoll``.
+If the app name of the ``Poll`` model is `election` then the history table name would be ``election_historicalpoll``.
+
+This feature allows a different and/or shorter model name and would even allow model names to be in the standard Django form
+
+    [appName]_[modelName]
+
+where the history models are created in a dedicated app using the "app" option thus making the resultant history table
+name much more aligned with the base table it is tracking
+
+To configure history models to use a different name for the history model class, use an option named ``custom_model_name``
+
+The value for this option can be a string or a callable.
+
+The most simple use case is illustrated below:
+
+.. code-block:: python
+
+    class ModelNameExample(models.Model):
+        history = HistoricalRecords(
+            custom_model_name='SimpleHistoricalModelNameExample'
+        )
+
+As an example using the callable mechanism, the below changes the default prefix `Historical` to `Audit`:
+
+.. code-block:: python
+
+    class Poll(models.Model):
+        question = models.CharField(max_length=200)
+        history = HistoricalRecords(custom_model_name=lambda x:f'Audit{x}')
+
+    class Opinion(models.Model):
+        opinion = models.CharField(max_length=2000)
+
+    register(Opinion, custom_model_name_prefix=lambda x:f'Audit{x}')
+    
+The resulting history class names would be `AuditPoll` and `AuditOpinion`.
+If the app the models are defined in is `yoda` then the corresponding history table names would be `yoda_auditpoll` and `yoda_auditopinion`
+ 
+You could also use a base class for your models and declare the history instantiation in the base class.
+All models that inherit the base model would automatically get this custom name.
+Using this mechanism requires using the `inherit` option.
+
+.. code-block:: python
+
+    class BaseModel(models.Model):
+        history = HistoricalRecords(inherit=True, app="yoda", custom_model_name_prefix=lambda x:f'Audit{x}')
+        class Meta:
+            abstract = True
+
+    class Poll(BaseModel):
+        question = models.CharField(max_length=200)
+
+    class Opinion(BaseModel):
+        opinion = models.CharField(max_length=2000)
+
+The resulting history class names would be `AuditPoll` and `AuditOpinion`.
+The history model tables will be named `yoda_auditpoll` and `yoda_auditopinion`.
+
+If the instantiation in the above example used `lambda x:f'{x}'` for the `custom_model_name` option
+  then the resulting history class names would be `HistoricalPoll` and `HistoricalOpinion`.
```

Comment:
Why doesn't this change the class names?

Reply:
Definitely want to keep the classnames consistent here. Otherwise I would focus strictly on table names

Reply:
Same as below comment on history model having identical name ot the model it is tracking ... can change this if you are happy it will not cause issues.

Reply:
I think we shouldn't allow this functionality. The model name has to be different

Reply:
Resolving this as it is the same as issue as the thread below